rust/extractor/src/qltest.rs

```diff
@@ -0,0 +1,84 @@
+use crate::config::Config;
+use anyhow::Context;
+use glob::glob;
+use itertools::Itertools;
+use log::info;
+use std::fs;
+use std::process::Command;
+
+fn dump_lib() -> anyhow::Result<()> {
+    let path_iterator = glob("*.rs").context("globbing test sources")?;
+    let paths = path_iterator
+        .collect::<Result<Vec<_>, _>>()
+        .context("fetching test sources")?;
+    let lib = paths
+        .iter()
+        .filter(|p| !["lib.rs", "main.rs"].contains(&p.file_name().unwrap().to_str().unwrap()))
+        .map(|p| format!("mod {};", p.file_stem().unwrap().to_str().unwrap()))
+        .join("\n");
+    fs::write("lib.rs", lib).context("writing lib.rs")
+}
+
+fn dump_cargo_manifest() -> anyhow::Result<()> {
+    let mut manifest = String::from(
+        r#"[workspace]
+[package]
+name = "test"
+version="0.0.1"
+edition="2021"
+[lib]
+path="lib.rs"
+"#,
+    );
+    if fs::exists("main.rs").context("checking existence of main.rs")? {
+        manifest.push_str(
+            r#"[[bin]]
+name = "main"
+path = "main.rs"
+"#,
+        );
+    }
+    fs::write("Cargo.toml", manifest).context("writing Cargo.toml")
+}
+
+fn set_sources(config: &mut Config) -> anyhow::Result<()> {
```

Comment:
Wouldn't it be better to let `codeql database index-files` take care of the globbing and just use the file-list argument as done in non-qltest extraction mode.

Reply:
`lib.rs` gets generated by the extractor itself, so that wouldn't be included... though maybe that could actually be better? I'll mull it over.

Reply:
let's maybe postpone this

Reply:
Sure, no problem.